core/src/main/kotlin/com/strumenta/kolasu/cli/CLI.kt

```diff
@@ -1,90 +0,0 @@
-import com.strumenta.kolasu.model.Node
-import com.strumenta.kolasu.model.children
-import com.strumenta.kolasu.parsing.KolasuParser
-import com.strumenta.kolasu.parsing.ParseTreeOrigin
-import com.strumenta.kolasu.parsing.ParsingResult
-import org.antlr.v4.runtime.ParserRuleContext
-import org.antlr.v4.runtime.tree.ParseTree
-import java.io.File
-import kotlin.system.exitProcess
-
-data class CLIContext<R : Node>(val language: KolasuParser<R, *, *>, var input: String? = null)
-
-open class ParserCLI<R : Node>(val parser: KolasuParser<R, *, *>) : CliktCommand() {
-    val input by argument()
-    val context by findOrSetObject { CLIContext(parser) }
-    override fun run() {
-        context.input = input
-    }
-}
-
-open class ParsingCommand<R : Node>(
-    help: String = "",
-    epilog: String = "",
-    name: String? = null,
-    invokeWithoutSubcommand: Boolean = false,
-    printHelpOnEmptyArgs: Boolean = false,
-    helpTags: Map<String, String> = emptyMap(),
-    autoCompleteEnvvar: String? = "",
-    allowMultipleSubcommands: Boolean = false,
-    treatUnknownOptionsAsArgs: Boolean = false,
-    val exitOnFail: Boolean = true,
-    val printASTSummary: Boolean = true
-) : CliktCommand(
-    help, epilog, name, invokeWithoutSubcommand, printHelpOnEmptyArgs, helpTags, autoCompleteEnvvar,
-    allowMultipleSubcommands, treatUnknownOptionsAsArgs
-) {
-    var result: ParsingResult<R>? = null
-    val context by requireObject<CLIContext<R>>()
-    override fun run() {
-        val input = File(context.input!!)
-        print("Parsing $input... ")
-        result = context.language.parse(input)
-        if (result!!.correct) {
-            println("OK.")
-        } else {
-            println("failed. Issues: ")
-            result!!.issues.forEach {
-                println('\t' + it.message + if (it.position != null) " @ ${it.position}" else "")
-            }
-            if (exitOnFail) {
-                exitProcess(2)
-            }
-        }
-        done()
-    }
-
-    open fun done() {
-        println(
-            "Done (${result!!.time}ms of which ${result!!.time!! - result!!.firstStage!!.time!!}ms to build the AST)."
-        )
-        val node = result!!.root!!
-        val position = node.position!!
-        val lines = position.end.line - position.start.line
-        val parseTree = (node.origin as ParseTreeOrigin).parseTree as ParserRuleContext
-        println(
-            "The file has $lines lines and ${result!!.code?.length ?: "unknown"} characters. " +
-                "The parse tree has ${countParseTreeNodes(parseTree)} nodes " +
-                "(${parseTree.start.inputStream.size()} tokens)."
-        )
-        if (printASTSummary) {
-            println("The AST has ${countASTNodes(node)} nodes.")
-        }
-    }
-
-    fun countASTNodes(node: Node): Int = 1 + (node.children.map { countASTNodes(it) }.reduceOrNull(Int::plus) ?: 0)
-
-    fun countParseTreeNodes(parseTree: ParseTree): Int {
```

Comment:
This part has been removed as in general I would tend to hide parse trees under the hood: we may not always have them. Perhaps instead I am wrong and these things are useful while developing parsers based on ANTLR